src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -129,6 +129,8 @@ class ModelCheckpoint(Checkpoint):
             where both values for ``every_n_epochs`` and ``check_val_every_n_epoch`` evenly divide E.
         save_on_train_epoch_end: Whether to run checkpointing at the end of the training epoch.
             If this is ``False``, then the check runs at the end of the validation.
+        overwrite_existing: Whether to overwrite the checkpoint file.
+            If this is ``True``, then the name of the saved file won't be appended with a version.
```

Comment:
I propose that we are explicit here and name this `enable_version_counter` (or similar)

overwrite could refer to different things (the input checkpoint, the output checkpoints, top_k checkpoints, save_last checkpoints, ...)

Reply:
@carmocca What else bothers me: besides the best checkpoint, the versioning approach is also applied to the [last one](https://github.com/Lightning-AI/lightning/blob/6b7bc421f26a724cf97c2b9d0109d583a0592972/src/lightning/pytorch/callbacks/model_checkpoint.py#L648). So if we stick to `enable_version_counter` does it make sense to update the `_save_last_checkpoint` as well?

Reply:
I guess that there is no need to separate them (at least for now, I can't come up with a case when we would need such functionality). You either overwrite files in the directory or create new ones by versioning.

Reply:
It's a stretch but maybe you want to version "top-k" checkpoints but not "last" checkpoints

Reply:
Ok, do you find `enable_top_k_versioning`, `enable_last_versioning` reasonable names for such params?

Reply:
No, just have one param that applies to both. For those who want to apply it to one, they'd do:

```python
mc1 = ModelCheckpoint(save_top_k=5, enable_version_counter=True)
mc2 = ModelCheckpoint(save_last=True, enable_version_counter=False)
trainer = Trainer(callbacks=[mc1, mc2])
```

Reply:
Oh, I see, that makes sense. I'll go with implementing `enable_version_counter`. Thanks!